### PR TITLE
Improve documentation for EventContext fields

### PR DESCRIPTION
**Pull request by richvdh**

### CODE REVIEW

File: synapse/events/snapshot.py [hunk 1]
```diff
@@ -12,52 +12,88 @@
...
+        rejected: A rejection reason if the event was rejected, else False
+
+        state_group: The ID of the state group for this event. Note that state events
+            are persisted with a state group which includes the new event, so this is
+            effectively the state *after* the event in question.
+
+            For a *rejected* state event, where the state of the rejected event is
+            ignored, this state_group should never make it into the
+            event_to_state_groups table. Indeed, inspecting this value for a rejected
+            state event is almost certainly incorrect.
```
Comment (erikjohnston) [inline synapse/events/snapshot.py]: (I wonder if we should make accessing `state_group` raise if its rejected)

Comment (richvdh) [reply synapse/events/snapshot.py]: yup could do!

File: synapse/events/snapshot.py [hunk 2]
```diff
@@ -12,52 +12,88 @@
...
+
+            For an outlier, where we don't have the state at the event, this will be
+            None.
+
+        prev_group: If it is known, ``state_group``'s prev_group. Note that this being
```
Comment (erikjohnston) [inline synapse/events/snapshot.py]: Might be worth noting that this is *not* the state group associated with `prev_state_ids`

File: synapse/events/snapshot.py (hunk 1 above)
Comment (richvdh) [reply synapse/events/snapshot.py]: well who knew, this identified a bug. I'm going to pull this out to a separate PR.

**Pull request merged**